src/poetry/utils/cache.py

```diff
@@ -196,3 +201,35 @@ def _deserialize(self, data_raw: bytes) -> CacheItem[T]:
         data = json.loads(data_str[10:])
         expires = int(data_str[:10])
         return CacheItem(data, expires)
+
+
+def get_cached_archives_for_link(cache_dir: Path, link: Link) -> list[Path]:
```

Comment:
Since all of these methods require the `cache_dir`, which will not change in one invocation of poetry, I think I'd like it better to introduce a class `ArtifactCache` that takes the cache directory in its `__init__()` and make all these functions instance methods of this class.